src/meltano/core/project_add_service.py

```diff
@@ -106,9 +112,12 @@ def add_related(self, *args, **kwargs):
         Returns:
             The added plugins.
         """
+        if project_plugin.requirements:
+            return self.add_required(project_plugin)
```

Comment:
`add_required` was never actually being called before, so the `required` property was being ignored, and `meltano add orchestrator airflow` would look for plugins with the same `namespace`, find `airflow` in `discovery.yml` (but not on the Hub because it had been renamed to `files-airflow` there), and try to install it. Then at some point it would apparently look for that plugin from `discovery.yml` on the Hub, and not find it there and fail with the error shown in https://github.com/meltano/meltano/issues/6046.

Now, if `required` is set, `namespace`-based related plugins are ignored.